### PR TITLE
poc time extrapolation functions

### PR DESCRIPTION
**Pull request by jmcook1186**
### Types of changes

- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### A description of the changes proposed in the Pull Request
Adds ability to extrapolate time series forwards and backwards in time to meet global start/end values by zero-padding.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,8 +13,11 @@
     "url": "https://github.com/Green-Software-Foundation/if/issues/new?assignees=&labels=feedback&projects=&template=feedback.md&title=Feedback+-+"
   },
   "dependencies": {
+    "@grnsft/if-models": "^0.1.2",
+    "@grnsft/if-unofficial-models": "^0.1.2",
```
Comment (narekhovhannisyan) [inline package.json:17]: these should not be commited

File: src/models/time-sync.ts [hunk 2]
```diff
@@ -136,15 +138,70 @@ export class TimeSyncModel implements ModelPluginInterface {
...
+        ),
+        moment(this.endTime)
+      );
+      console.log('DATE RANGE ', dateRange);
+      console.log(inputs);
+      for (const second of dateRange.by('second')) {
+        // @todo apply zero-fill logic to create this object (any fields with aggregation-method == sum or avg should be zeros, others copied from inputs[0])
+        inputs.push({timestamp: second.toISOString(), duration: 1, carbon: 0});
+      }
+      inputs.sort((a, b) => moment(a.timestamp).diff(moment(b.timestamp)));
```
Comment (narekhovhannisyan) [inline src/models/time-sync.ts:189]: sorting is not applied to inputs since returned value is not used

Comment (jmcook1186) [reply src/models/time-sync.ts]: oh good point - fixing now

File: package.json (hunk 1 above)
Comment (jmcook1186) [reply package.json:17]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -13,8 +13,11 @@
     "url": "https://github.com/Green-Software-Foundation/if/issues/new?assignees=&labels=feedback&projects=&template=feedback.md&title=Feedback+-+"
   },
   "dependencies": {
+    "@grnsft/if-models": "^0.1.2",
```
Comment (jmcook1186) [inline package.json:16]:
```suggestion
```

**Pull request merged**